frontend/entities/Gateways.md

```diff
@@ -0,0 +1,20 @@
+# Gateways
+
+The gateways encapsulate the I/O logic that will be used across the application, as well as Services, these are static modules that provide a list of functions.
+As part of the gateway job is to generate the model instances based on the response as well as error handling.
+
+```ts
+const ImageGateway = () => ({
+  async getUserById(userId: string) {
+    const response = await fetch(path.replace('{{userId}}', userId));
+
+    if (!response.ok) throw Error('user not found')
+
+    const rawUser = await response.json();
+
+    return User(rawUser);
```

Comment:
What happens if rawUser does not contain the correct data to instantiate a User?

Reply:
It is part of the model task, take a look a the first point from that file